src/formatMarkup.js

```diff
@@ -75,15 +81,15 @@ export const diffableFormatter = function (markup, options) {
       lastSeenTag = node.tagName;
     }
 
-    const tagIsWhitespaceDependent = WHITESPACE_DEPENDENT_TAGS.includes(lastSeenTag);
+    const tagIsWhitespaceDependent = (Array.isArray(options.tagsWithWhitespacePreserved) && options.tagsWithWhitespacePreserved.includes(lastSeenTag));
```

Comment:
These two variables (`tagIsWhitespaceDependent` & `areAllTagsWhiteSpacePreserved`), are only ever used together, and are really checking for the same thing, so it's simpler to combine them.
```suggestion
    const tagIsWhitespaceDependent = (
      options.tagsWithWhitespacePreserved === true ||
      (
        Array.isArray(options.tagsWithWhitespacePreserved) &&
        options.tagsWithWhitespacePreserved.includes(lastSeenTag)
      )
    );
```